### PR TITLE
Feature/bloom filter for dict

### PR DESCRIPTION
**Pull request by TangoBeeAkto**

### CODE REVIEW

File: libs/dao/src/test/java/com/akto/utils/filter/TestDictionaryFilter.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    @Test
+    public void testEmptyString() {
+        assertFalse(DictionaryFilter.isEnglishWord(""));
+    }
+
+    @Test
+    public void testMixedCaseWords() {
+        assertTrue(DictionaryFilter.isEnglishWord("Demo"));
+        assertTrue(DictionaryFilter.isEnglishWord("CaT"));
+        assertFalse(DictionaryFilter.isEnglishWord("NotExist"));
```
Comment (avneesh-akto) [inline libs/dao/src/test/java/com/akto/utils/filter/TestDictionaryFilter.java:59]: this should return true

File: apps/api-runtime/src/main/java/com/akto/runtime/Main.java [hunk 2]
```diff
@@ -157,6 +157,8 @@ public static void main(String[] args) {
         }
         int maxPollRecordsConfig = Integer.parseInt(System.getenv("AKTO_KAFKA_MAX_POLL_RECORDS_CONFIG"));
 
+        DictionaryFilter.readDictionaryBinary();
```
Comment (avneesh-akto) [inline apps/api-runtime/src/main/java/com/akto/runtime/Main.java:160]: do for mini runtime

File: apps/mini-runtime/src/main/java/com/akto/hybrid_runtime/APICatalogSync.java [hunk 3]
```diff
@@ -325,6 +313,8 @@ private void tryMergingWithKnownStrictURLs(
         while (iterator.hasNext()) {
             Map.Entry<URLStatic, RequestTemplate> entry = iterator.next();
             URLStatic newUrl = entry.getKey();
+            if(DictionaryFilter.isEnglishWord(newUrl.getUrl())) continue;
```
Comment (avneesh-akto) [inline apps/mini-runtime/src/main/java/com/akto/hybrid_runtime/APICatalogSync.java:316]: fix this

File: apps/mini-runtime/src/main/java/com/akto/hybrid_runtime/APICatalogSync.java [hunk 4]
```diff
@@ -636,6 +626,8 @@ private void tryWithKnownURLTemplates(
             while (iterator.hasNext()) {
                 Map.Entry<URLStatic, RequestTemplate> entry = iterator.next();
                 URLStatic newUrl = entry.getKey();
+                if(DictionaryFilter.isEnglishWord(newUrl.getUrl())) continue;
```
Comment (avneesh-akto) [inline apps/mini-runtime/src/main/java/com/akto/hybrid_runtime/APICatalogSync.java:629]: fix this

File: apps/api-runtime/src/main/java/com/akto/runtime/APICatalogSync.java [hunk 5]
```diff
@@ -768,6 +764,7 @@ public static URLTemplate tryMergeUrls(URLStatic dbUrl, URLStatic newUrl) {
         for(int i = 0; i < newTokens.length; i ++) {
             String tempToken = newTokens[i];
             String dbToken = dbTokens[i];
+            if (DictionaryFilter.isEnglishWord(tempToken) && DictionaryFilter.isEnglishWord(dbToken)) continue;
```
Comment (avneesh-akto) [inline apps/api-runtime/src/main/java/com/akto/runtime/APICatalogSync.java:767]: or instead of and

**Review by avneesh-akto — APPROVED** [review]

File: libs/dao/src/main/java/com/akto/util/filter/DictionaryFilterCreator.java [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+    public static void main(String[] args) {
+        DictionaryFilterCreator.insertDictionary();
+//        DictionaryFilterCreator.readDictionary();
+    }
+    public static BloomFilter<CharSequence> dictFilter = BloomFilter.create(Funnels.stringFunnel(StandardCharsets.UTF_8), 1_000_000, 0.001);
+
+    public static void insertDictionary() {
+        /*
+            - English Word List Repo: https://github.com/dwyl/english-words/blob/master/words_alpha.txt
```
Comment (TangoBeeAkto) [inline libs/dao/src/main/java/com/akto/util/filter/DictionaryFilterCreator.java:20]: update with correct akto resources path

File: libs/dao/src/main/java/com/akto/util/filter/DictionaryFilter.java [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+                break;
+            }
+        }
+
+        if(flag) return true;
+
+        return dictFilter.mightContain(word);
+    }
+
+    private static boolean wordsChecker(String[] words) {
```
Comment (TangoBeeAkto) [inline libs/dao/src/main/java/com/akto/util/filter/DictionaryFilter.java:47]: add comment to explain

File: libs/dao/src/main/java/com/akto/dto/filter/MergedUrls.java [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+    private int apiCollectionId;
+
+    public MergedUrls() {}
+
+    public MergedUrls(String url, String method, int apiCollectionId) {
+        this.url = url;
+        this.method = method;
+        this.apiCollectionId = apiCollectionId;
+    }
+
```
Comment (TangoBeeAkto) [inline libs/dao/src/main/java/com/akto/dto/filter/MergedUrls.java:24]: add hash and equal function

**Pull request merged**